### PR TITLE
Deprecate Joomla.loadingLayer

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue # .

### Summary of Changes
Deprecating the spinner creator in core.js. Joomla 4 will introduce a web component for this.

So in 4.0 in each page that a spinner is needed the following steps will need to be done:

- Include the web component
```php
HTMLHelper::_('webcomponent', 'system/webcomponents/joomla-core-loader.min.js', ['relative' => true, 'version' => 'auto']);
```

- To show the spinner all we need to do is create an element:
```js
spinner = document.createElement('joomla-core-loader');

// Assuming that the spinner is full screen
document.body.appendChild(spinner);

/*
if we need to append the spinner to some other container
all we have to do is replace document.body with the element 
that we want the spinner to appear
*/
```

- To remove the spinner all we need to do is destroy the element:
```js
spinner = document.querySelector('joomla-core-loader');

spinner.parentNode.removeChild(spinner);
```

### Documentation Changes Required

@wilsonge @mbabker

### CODE REVIEW

File: media/system/js/core-uncompressed.js [hunk 1]
```diff
@@ -775,6 +775,10 @@ Joomla.editors.instances = Joomla.editors.instances || {
 	 * @return  {HTMLElement}  The HTML loading layer element.
 	 *
 	 * @since  3.6.0
+	 *
+	 * @deprecated  4.0 No direct replacement.
+	 *              4.0 will introduce a web component for the loading spinner, therefore the spinner will need to
+	 *              explicitly loaded in all relevant pages.
```
Comment (brianteeman) [inline media/system/js/core-uncompressed.js]: explicitly be loaded

Comment (dgrammatiko) [reply media/system/js/core-uncompressed.js]: Thanks

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on b6007715ec1bf71eee1356c6a9019931107eeaf9<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21460">issues.joomla.org/tracker/joomla-cms/21460</a>.</sub>

**Comment by laoneo** [comment]
Will the old way also work in J4?

**Comment by dgrammatiko** [comment]
> Will the old way also work in J4?

No

**Comment by laoneo** [comment]
And I assume the new way will also not work in J3. So what does an extension dev need to do then? In PHP you can write a version compare check and then execute the relevant code depending n the major version. How do we do that on the JS side? The target is that J3 extensions will also work in J4 if they are up to date.

**Comment by dgrammatiko** [comment]
> So what does an extension dev need to do then?

I provided the lines above the missing step, lets say if you have crap build tools and you use the same codebase for different versions is quite simple:

```js
if (Joomla.loadingLayer && typeof Joomla.loadingLayer === 'function') {
 // We are in J3 so use the old method
} else {
 // We are in the future
spinner = document.createElement('joomla-core-loader');
document.body.appendChild(spinner);
}
```

So basically it's almost the same as the JVersion trick for the PHP part.

**Review by laoneo — APPROVED** [review]

**Comment by laoneo** [comment]
I have tested this item :white_check_mark: successfully on b6007715ec1bf71eee1356c6a9019931107eeaf9<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21460">issues.joomla.org/tracker/joomla-cms/21460</a>.</sub>

**Comment by ghost** [comment]
Ready to Commit after two successful tests.

**Pull request merged**